### PR TITLE
feat(client): enhance WhereClauseCanonicalizer 

### PR DESCRIPTION
**Pull request by ericanderson**
## Summary

This PR enhances the `WhereClauseCanonicalizer` in the observable client with advanced optimization capabilities to improve query performance and consistency.

### Key Enhancements

- **Nested clause flattening**: Automatically flattens nested `$and` and `$or` clauses to create simpler, more efficient query structures
- **Property filter merging**: Merges multiple property filters within `$and` clauses to reduce redundancy and improve performance
- **Double negation elimination**: Eliminates double negations in `$not` clauses (e.g., `$not { $not: X }` becomes `X`)
- **Deterministic canonicalization**: Ensures consistent sorting of clauses for predictable and deterministic query canonicalization
- **Comprehensive testing**: Adds extensive test coverage for all optimization scenarios including edge cases

### Technical Details

The implementation adds three new private helper methods:
- `#flattenAndMergeAnd()`: Handles flattening nested `$and` clauses and merging property filters
- `#mergeIntoPropertyMap()`: Merges simple property filters into a single object for efficiency
- `#flattenOr()`: Recursively flattens nested `$or` clauses

All optimizations maintain semantic equivalence while improving query structure and performance.

### Test Coverage

Added 271+ lines of comprehensive test cases covering:
- Nested clause flattening for both `$and` and `$or` operations
- Property filter merging within `$and` clauses
- Double negation elimination scenarios
- Complex nested structure canonicalization
- Edge cases like empty arrays and single-element arrays

## Test plan

- [x] All existing tests pass
- [x] New comprehensive test suite validates all optimization scenarios
- [x] Tests ensure semantic equivalence is maintained after optimization
- [x] Performance improvements verified through canonicalization benchmarks

### CODE REVIEW

File: packages/client/src/observable/internal/WhereClauseCanonicalizer.test.ts [hunk 1]
```diff
@@ -160,4 +160,271 @@ describe(WhereClauseCanonicalizer, () => {
...
+      ],
+    };
+
+    const r1 = c.canonicalize(w1);
+
+    // Should merge employeeId and class into single property filter
+    expect(r1).toEqual({
+      $and: [
+        { $or: [{ fullName: "Jane" }, { fullName: "John" }] }, // complex clause comes first due to JSON sorting
+        { employeeId: 5, class: "what" }, // merged properties
```
Comment (ssanjay1) [inline packages/client/src/observable/internal/WhereClauseCanonicalizer.test.ts:210]: Probably worth adding a test for when you have something like 
```
 $and: [
        { employeeId: {$gt:5} },
        { employeeId: {$lt:8} },
        { $or: [{ fullName: "John" }, { fullName: "Jane" }] },
      ],
      ```
      Where the properties you need to merge are the same
